pliers/extractors/text.py

```diff
@@ -402,3 +400,140 @@ def _extract(self, stim):
+            If different from model, out-of-vocabulary tokens may be treated as
+            unknown tokens.
+        framework (str): name deep learning framework to use. Must be 'pt'
+            (PyTorch) or 'tensorflow'. Defaults to 'pt'.
+        encoding_level (str): A string specifying whether encodings for each
+            token or sequence encodings are to be returned. Must be one of
+            'token', 'sequence'.
+        pooling (str): Optional argument, relevant for sequence-level
+            embeddings only. If None and encoding_level='sequence', encodings
+            for [CLS] tokens are returned. If encoding_level='sequence' and
+            numpy function is specified, token-level embeddings are pooled
+            according to specified method (e.g. 'mean', 'max', 'min').
+        model_kwargs (dict): Named arguments for pretrained model.
+            See: https://huggingface.co/transformers/main_classes/model.html
+            and https://huggingface.co/transformers/model_doc/bert.html for
+            further info.
+        tokenizer_kwargs (dict): Named arguments for tokenizer.
+            See https://huggingface.co/transformers/main_classes/tokenizer.html
+            for further info.
+    '''
+
+    _log_attributes = ('pretrained_model', 'encoding_level',
+                       'pooling', 'framework', 'tokenizer_type',
+                       'model_kwargs', 'tokenizer_kwargs')
+
+    def __init__(self,
+                 pretrained_model_or_path='bert-base-uncased',
+                 tokenizer='bert-base-uncased',
+                 framework='pt',
+                 encoding_level='token',
+                 pooling=None,
+                 model_kwargs=None,
+                 tokenizer_kwargs=None):
+                 
+        verify_dependencies(['transformers'])
+
+        if framework not in ['pt', 'tf']:
+            raise(ValueError('''Invalid framework;
+                must be one of 'pt' (pytorch) or 'tf' (tensorflow)'''))
+
+        if model_kwargs is None:
+            model_kwargs = {}
+
+        if tokenizer_kwargs is None:
+            tokenizer_kwargs = {}
+
+        self.pretrained_model = pretrained_model_or_path
+        self.tokenizer_type = tokenizer
+        self.framework = framework
+        self.encoding_level = encoding_level
+        self.pooling = pooling
+        self.model_kwargs = model_kwargs
+        self.tokenizer_kwargs = tokenizer_kwargs
+        model_name = 'BertModel' if self.framework == 'pt' else 'TFBertModel'
+
+        self.tokenizer = transformers.BertTokenizer.from_pretrained(
+            tokenizer, **tokenizer_kwargs)
+        self.model = getattr(transformers, model_name).from_pretrained(
+            pretrained_model_or_path, **model_kwargs)
+
+        super(PretrainedBertEncodingExtractor, self).__init__()
+
+    def _extract(self, stims):
+
+        stims.name = ' '.join([s.text for s in stims.elements])
```

Comment:
I have mixed feelings about this... I assume the point is to have a string that identifies the `ComplexTextStim`, which makes sense... But the user may have their own reason for setting the `.name` property. Then again, they'd currently have to set that directly, as we don't pass a field through from the `ComplexTextStim` initializer (which is actually a bit of an oversight). So, not sure...

Reply:
would adding an "input_sequence" field to log attributes be a viable solution (only returned in result df if model_attributes=True in `_to_df`)?

Reply:
`_log_attributes` applies at the transformer instance level, not the stim passed to `transform()`, so I don't think that would work.

Maybe we just bite the bullet and add an `input_sequence` column to the returned DF. This would mess things up a bit in long format, but so be it. (Probably what we should really do is rework the `to_df` logic so that there's a way to tag columns as index vs. content columns.)

Reply:
TBH I'm thinking it may have been a bad idea to support the `format` argument in the first place. A lot of the complexities we've faced are introduced by that distinction, and I think it would be pretty reasonable to always return `'wide'` and leave it to users to do any reshaping they need to.

Either way, let's not worry about it in this PR—you can make an executive decision about whether to overwrite `.name` or add a column, and we can revisit the more general issue later.

Reply:
ok, sounds like this is a more general problem indeed. I've added a conditional statement which overwrites `.name` only if it is an empty string (leaving some freedom to the user to specify it). It's just a temporary patch, but might do the job until we decide what to do with `to_df`.